### ISSUE TITLE
Common format?

### ISSUE THREAD

**Issue by frederic-santos** [ess-intro/ess-intro#3]
Dirk @eddelbuettel  wrote in the wiki:

> Markdown is great for slides and other writeups. We will work out which, if any, common slide format (or formats as we may support both html and pdf) we use.

Is there a common format, or even template, that we should use? I'm not a heavy user of rmarkdown and usually prefer to write in org mode, for instance. 

1. Is it okay for the docs to be written in org mode?
2. Ideally, should they be exported as beamer slides, reveal.js slides, classical pdf/LaTeX document, or something else?

**Comment by eddelbuettel** [comment]
I wondered about that too. I use `binb` a lot (it is a beamer 'metropolis' / 'mtheme' variant) and it produces pdf; there is a xaringan theme that looks similar and produces html.  Open to suggestions.

**Comment by plantarum** [comment]
It might be helpful to have a clearer idea of what the end product is going to be, and maybe also a timeline/deadline? Having a common format is a nice idea, but it suggests this is becoming a more formal project than I had anticipated (not a bad thing, just different!). 

I've been expecting to do a live demo via Zoom/Meet/whatever platform at some point. This could also be a recorded screen cast, I haven't looked at the platform Dirk suggested yet, but I'm happy to have the opportunity to learn how to record screencasts.

The notes I've prepared are intended to serve both as stand-alone written tutorials, and things I can reference in my demo so I don't have to spend much time on configuration specifics. They're also test cases for the demos themselves - I run the tutorial code in `emacs -Q` to make sure I include all the necessary installation/config details for the features I demo. 

That being the case, I'd be happy converting my Rmd sources to org format if we want to standardize on that. I will be creating a separate skeleton document to walk through for my demo, but I hadn't been planning on making slides.

**Comment by eddelbuettel** [comment]
Well as mentioned a few times I had something "simple" and "easy" in mind following what I had done last spring with the t^4 series on command-line tricks (for R) from which the "let's do Emacs!" suggestion arose too.  That used a short video (posted to YouTube!, folks really seemed to like these) as the main vehicle supported by slides (used in the video, also for separate consumption and notes and links etc).  By using Rmd input it should be pretty trivial to switch look and feel of presentations.  The key are the short recorded talks around the slides.  Recording Zoom may work; I had used obs (which exists for all OSs).

The main thing, really, is to agree on a common or common-ish format.  For a large group of people full unamity of opinion is not expected to this may be fluid.  Ditto for timelines:  "when ready" is probably the only reasonable model.  So with that I think we should just work on the two or so intro topics and take it from there.  

Thoughts?

**Comment by plantarum** [comment]
That all sounds fine to me. I'm pretty excited that this is happening, and happy to contribute however. Both org and Rmd are easy enough to 'style'.

I have some time this month, so will continue on as I'm able. If we do decide on more defined style guidelines I'll do my best to accommodate.

**Comment by frederic-santos** [comment]
Actually if the guidelines only apply to the final ouput, I guess that the choice between Org and Rmd (almost) does not matter for the source files. Being able to use Org would be ideal for me, but it's fine as well if we decide to produce xaringan or binb outputs for all presentations.

Anyway, the combination written tuto (as slides) + live demo is really good, and both are useful. For the 'debugging with ESS' topic for instance, both of them are necessary. So this is really a great project and I'm happy to contribute too.

**Comment by eddelbuettel** [comment]
Yes, I see org-mode as a non-issue as Rmd still flows from it we get to that common entry point. 

All this *is* quite exciting and maybe GitHub issues here also pan out for "general" discussions.

**Comment by aezarebski** [comment]
I've added a section to the Style wiki page summarising some of this.

https://github.com/ess-intro/ess-intro/wiki/Style#technology-for-producing-presentations

**Comment by frederic-santos** [comment]
Thanks!

Just to confirm (better to ask before going too far in the process), is this okay? https://github.com/ess-intro/presentation-ess-customization
The tutorial can be read either on Github as an org file, or as PDF (beamer) slides. If you all prefer HTML slides instead (although I think that a PDF output is useful if one wants to print the tuto), or in addition, I'll change that.

Also, feel free to re-use the beamer template (with the "official" purple Emacs colors :-)) or to suggest modifications.

**Comment by eddelbuettel** [comment]
`org-mode` to `beamer` seems to be a standard way so thumbs up (and I had not realized GitHub already typesets .org files so nicely).  

Can you maybe add a `Makefile` (or a script) nailing down the conversion steps?

**Comment by aezarebski** [comment]
Seeing as we have a volunteer for an org-mode presentation, @greg-minshall , perhaps they might have some wisdom to offer. For those starting from rmarkdown @plantarum might have some tips. 

I've never managed to get comfortable with beamer so would appreciate any tips.

**Comment by eddelbuettel** [comment]
> I've never managed to get comfortable with beamer so would appreciate any tips.

Maybe specifics could/should go into a new issue?  But e.g. I switched five or so years ago from plain beamer (and latex) to mtheme/metropolis first (still have a (deprecated) "sample" repo for that [here](https://github.com/eddelbuettel/samples-rmarkdown-metropolis) as well as old repo packaging metropolis when it wasn't in texlive yet [here](https://github.com/eddelbuettel/pkg-latex-metropolis)) but now use `binb`  (on [CRAN](http://cloud.r-project.org/package=binb) and [GitHub](https://github.com/eddelbuettel/binb)) directly via Rmd.  `binb` contains three other nice styles too, all work from RMarkdown.

The html crowd seems to like [xaringan](https://cran.r-project.org/package=xaringan) which can be made to look like metropolis too.

**Comment by greg-minshall** [comment]
n/c.  i've not used org ==> markdown export, so have no idea how well that would please a markdown person, let alone a tool that goes from markdown to html5, etc.  (i mostly use org to hold a bunch of source code and do a small amount of exporting to html or pdf.)

**Comment by plantarum** [comment]
I've been working on the RMarkdown/polymode part of my presentation. When I get around to exporting, I'll review the options for getting beamer output, vs html-based slides. I've done both in the past, but until I finish my review I won't know what's easiest with the minimal configuration for beginners. 

(my own set up formed through accretion over many years, so this is a useful and occasionally quite surprising process to work through!)
